src/Symfony/Component/Cache/Adapter/OpCacheAdapter.php

```diff
@@ -0,0 +1,174 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Adapter;
+
+use Symfony\Component\Cache\Exception\InvalidArgumentException;
+
+/**
+ * Adapter building static PHP files that will be cached by OPCache.
+ * This Adapter is in read-only if you use AdapterInterface methods.
+ * You can use the method "warmUp" to build the cache file.
+ *
+ * @author Titouan Galopin <galopintitouan@gmail.com>
+ */
+class OpCacheAdapter extends AbstractAdapter
+{
+    private $file;
+    private $storeSerialized;
+    private $loaded;
+    private $values;
+
+    public function __construct($file, $namespace = '', $defaultLifetime = 0, $storeSerialized = true)
+    {
+        parent::__construct($namespace, $defaultLifetime);
+
+        $directory = dirname($file);
+
+        if (!file_exists($directory)) {
+            @mkdir($directory, 0777, true);
+        }
+
+        if (!file_exists($directory)) {
+            throw new InvalidArgumentException(sprintf(
+                'Cache directory does not exist and cannot be created (%s)',
+                $directory
+            ));
+        }
+
+        if (!is_writable($directory)) {
+            throw new InvalidArgumentException(sprintf('Cache directory is not writable (%s)', $directory));
+        }
+
+        $this->file = $file;
+        $this->storeSerialized = $storeSerialized;
+        $this->loaded = false;
+        $this->values = [];
+    }
+
+    public function warmUp(array $values = [])
+    {
+        // On PHP 5.6, OPcache is able to keep statically declared arrays in shared memory,
+        // but only up to 32767 elements (See https://bugs.php.net/68057).
+        // On PHP 7.0+, all static arrays are kept in shared memory, statically declared or not.
+        $static = count($values) > 32767 ? '' : 'static ';
+
+        if ($this->storeSerialized) {
+            foreach ($values as $id => $value) {
+                $values[$id] = serialize($value);
+            }
+        }
+
+        $exported = var_export($values, true);
+
+        $dump = <<<EOF
+<?php
+
+/**
+ * This file has been auto-generated
+ * by the Symfony Cache Component.
+ */
+ 
+$static\$staticArray = $exported;
+
+\$array =& \$staticArray;
+unset(\$staticArray);
+
+return \$array;
+EOF;
+
+        file_put_contents($this->file, $dump);
+
+        $this->loaded = false;
+        $this->loadFile();
```

Comment:
allows validating the generated file, especially useful in non-serialized mode where var_export can generate non loadable content